spec/components/search_spec.rb

```diff
@@ -703,6 +703,25 @@ def search
       expect(Search.execute('green tags:eggs').posts.map(&:id)).to eq([post2.id])
       expect(Search.execute('green tags:plants').posts.size).to eq(0)
     end
+
+    it "can find posts which contains filetypes" do
+      post1 = Fabricate(:post,
+                        raw: "http://example.com/image.png")
+      post2 = Fabricate(:post,
+                         raw: "Discourse logo\n"\
+                              "http://example.com/logo.png\n"\
+                              "http://example.com/vector_image.svg")
+      post_with_upload = Fabricate(:post)
+      post_with_upload.uploads = [Fabricate(:upload)]
+      Fabricate(:post)
+
+      TopicLink.extract_from(post1)
+      TopicLink.extract_from(post2)
+
+      expect(Search.execute('filetype:svg').posts.map(&:id)).to eq([post2.id])
```

Comment:
Why do we need to map the `id` here? I believe comparing against the post record should work just fine.